yarpl/include/yarpl/Refcounted.h

```diff
@@ -318,14 +318,32 @@ Reference<CastTo> make_ref(Args&&... args) {
   );
 }
 
-template <typename T>
-Reference<T> get_ref(T* object) {
-  static_assert(
-      std::is_base_of<Refcounted, std::decay_t<T>>::value,
-      "Reference can only be constructed with a Refcounted object");
+class enable_get_ref {
```

Comment:
I believe linter will complain that you should add a virtual dtor for this class... can you disable the warning?